src/Microsoft.Win32.SystemEvents/src/Microsoft/Win32/SystemEvents.cs

```diff
@@ -793,27 +660,64 @@ private UserPreferenceCategory GetUserPreferenceCategory(int msg, IntPtr wParam,
             return pref;
         }
 
-        private void Initialize()
+        private unsafe void Initialize()
         {
             _consoleHandler = new Interop.Kernel32.ConsoleCtrlHandlerRoutine(ConsoleHandlerProc);
-            if (!Interop.Kernel32.SetConsoleCtrlHandler(_consoleHandler, true))
+            _consoleHandlerPointer = Marshal.GetFunctionPointerForDelegate(_consoleHandler);
+
+            if (!Interop.Kernel32.SetConsoleCtrlHandler(_consoleHandlerPointer, true))
             {
                 Debug.Fail("Failed to install console handler.");
                 _consoleHandler = null;
             }
 
-            _windowHandle = CreateBroadcastWindow();
-            Debug.WriteLineIf(_windowHandle == IntPtr.Zero, "CreateBroadcastWindow failed");
+            IntPtr hInstance = Interop.Kernel32.GetModuleHandle(null);
+
+            s_className = string.Format(
+                ".NET-BroadcastEventWindow.{0:x}.0",
+                AppDomain.CurrentDomain.GetHashCode());
```

Comment:
This will box the hash code Int32.  If you're concerned about allocations, since there's only one appdomain, is this necessary, rather than just hardcoding a value?

Reply:
I considered hard coding it but held back on it as multiple apps could have the exact same Window name floating around and some obscure code out there might be looking for this. If we do hardcode it I would look at hardcoding something that never occurs on desktop.

Reply:
> I considered hard coding it but held back on it as multiple apps could have the exact same Window name floating around and some obscure code out there might be looking for this. If we do hardcode it I would look at hardcoding something that never occurs on desktop.

Won't the AppDomain hash codes be different from what they were on desktop?

Reply:
They are different for different apps and instances. I'm not opposed to hard coding it, I'm just trying to keep the same variability out of an abundance of caution on my part.